src/anlz.rs

```diff
@@ -359,8 +261,11 @@ pub struct ExtendedCue {
     pub loop_numerator: u16,
     /// Represents the loop size denominator (if this is a quantized loop).
     pub loop_denominator: u16,
-    /// And UTF-16BE encoded string, followed by a trailing  `0x0000`.
-    pub comment: String,
+    /// Length of the comment string in bytes.
+    pub len_comment: u32,
+    /// An UTF-16BE encoded string, followed by a trailing  `0x0000`.
+    #[br(count = len_comment)]
+    pub comment: Vec<u8>,
```

Comment:
Hmm, I'm unsure how to handle it properly. We need to make sure that we exactly parse that many bytes.

Reply:
Do you mean we can't rely on the fact that the string is null-terminated or that we need to make sure that the parsed string was only that many bytes long?
For the latter cased, an `assert` is likely sufficient, for the former case, I used something like this for the `pdb`:
```rust
UCS2(
  #[br(count = len / 2)]
  #[br(try_map = |x: Vec<u16>| String::from_utf16(&x))]
  #[bw(map = |x: &String| x.encode_utf16().collect::<Vec<u16>>())]
  String
)
```
Note: unfortunately using `&String` is required otherwise the macro trips up somehow.

We should probably factor this out into a separate type similar to `NullString` and `NullWideString` and then maybe file a PR to binrw so they get included. This case seems common enough.

Reply:
> Do you mean we can't rely on the fact that the string is null-terminated or that we need to make sure that the parsed string was only that many bytes long?
> For the latter cased, an assert is likely sufficient, for the former case, I used something like this for the pdb: [...]

I used a `NullWideString` for now. In theory, there should always be a nullbyte. The issue is that I always learned "do not trust user data". Depending on a nullbyte seems wrong because a maliciously crafted file may omit this and lead to freeze/OOM issues or something like that. And that would be unnecessary because we *know* beforehand how many bytes should be read.

IMHO the cleanest way would be to introduce a new attribute that limits the maximum number of bytes that can be read. That would be reusable even for other kinds of data structures.

```rust
pub struct Foo {
    len_text: u32,
    #[br(limit = len_text)]
    text: NullString,
}

pub struct Bar {
    #[br(assert(len_text % 2 == 0))]
    len_text: u32,
    #[br(limit = len_text)]
    text: NullWideString,
}